packages/dynamics-lib/src/entities/__tests__/pocl-data-validaton-error.entity.spec.js

```diff
@@ -0,0 +1,157 @@
+        defra_town: 'Bristol',
+        defra_postcode: 'BS9 1HJ',
+        defra_country: 'GB',
+        defra_birthdate: '1989-07-01',
+        defra_emailaddress: 'daniel-ricc@example.com',
+        defra_mobilenumber: '07722 123456',
+        defra_preferredmethodofnewsletter: 910400003,
+        defra_preferredmethodofconfirmation: 910400000,
+        defra_preferredmethodofreminder: 910400002,
+        defra_concessions: '[{"type":"Blue Badge","referenceNumber":"123456789"}]',
+        defra_startdate: '2021-06-15',
+        defra_serialnumber: '14345-48457J',
+        defra_permitid: 'test-permit-id',
+        defra_transactiondate: '2020-01-01T14:00:00Z',
+        defra_amount: 30,
+        defra_paymentsource: 'Post Office Sales',
+        defra_channelid: '948594',
+        defra_methodofpayment: 910400001,
+        defra_status: 910400000,
+        defra_datasource: 910400000
+      },
+      optionSetData
+    )
+
+    const expectedFields = {
+      id: '91f15d18-0aa4-ea11-a812-000d3a64905b',
+      firstName: 'Daniel',
+      lastName: 'Ricciardo',
+      organisation: 'Fishy Endeavours',
+      premises: '14 Howecroft Court',
+      street: 'Eastmead Lane',
+      town: 'Bristol',
+      postcode: 'BS9 1HJ',
+      country: 'GB',
+      birthDate: '1989-07-01',
+      email: 'daniel-ricc@example.com',
+      mobilePhone: '07722 123456',
+      preferredMethodOfNewsletter: expect.objectContaining({
+        id: 910400003,
+        label: 'Prefer not to be contacted',
+        description: 'Prefer not to be contacted'
+      }),
+      preferredMethodOfConfirmation: expect.objectContaining({
+        id: 910400000,
+        label: 'Email',
+        description: 'Email'
+      }),
+      preferredMethodOfReminder: expect.objectContaining({
+        id: 910400002,
+        label: 'Text',
+        description: 'Text'
+      }),
+      startDate: '2021-06-15',
+      serialNumber: '14345-48457J',
+      permitId: 'test-permit-id',
+      concessions: '[{"type":"Blue Badge","referenceNumber":"123456789"}]',
+      transactionDate: '2020-01-01T14:00:00Z',
+      amount: 30,
+      paymentSource: 'Post Office Sales',
+      channelId: '948594',
+      methodOfPayment: expect.objectContaining({
+        id: 910400001,
+        label: 'Cash',
+        description: 'Cash'
+      }),
+      dataSource: expect.objectContaining({
+        id: 910400000,
+        label: 'Post Office Sales',
+        description: 'Post Office Sales'
+      }),
+      status: expect.objectContaining({ id: 910400000, label: 'Needs Review', description: 'Needs Review' })
+    }
+
+    expect(exception).toBeInstanceOf(PoclDataValidationError)
+    expect(exception).toMatchObject(expect.objectContaining({ etag: 'W/"56351087"', ...expectedFields }))
+    expect(exception.toJSON()).toMatchObject(expect.objectContaining(expectedFields))
+    expect(JSON.parse(exception.toString())).toMatchObject(expect.objectContaining(expectedFields))
```

Comment:
Multiple expect statements should generally be avoided in favour of several tests. So, have a test checking instance type, then a snapshot test (see below about those). In fact, looks like the next three expects could be replaced by a single snapshot test.

That said, if you do multiple expect statements as the data is taking a lot of mocking, then just have an external function that generates the test data, that you can use in each of your tests